be/src/exec/aggregator.cpp

```diff
@@ -820,6 +819,51 @@ Status Aggregator::output_chunk_by_streaming(Chunk* input_chunk, ChunkPtr* chunk
     return Status::OK();
 }
 
+Status Aggregator::convert_to_spill_format(Chunk* input_chunk, ChunkPtr* chunk) {
+    auto use_intermediate_as_input = _use_intermediate_as_input();
+    size_t num_rows = input_chunk->num_rows();
+    ChunkPtr result_chunk = std::make_shared<Chunk>();
+    const auto& slots = _intermediate_tuple_desc->slots();
+    // build group by column
+    for (size_t i = 0; i < _group_by_columns.size(); i++) {
+        DCHECK_EQ(num_rows, _group_by_columns[i]->size());
+        // materialize group by const columns
+        if (_group_by_columns[i]->is_constant()) {
+            auto res =
+                    ColumnHelper::unfold_const_column(_group_by_types[i].result_type, num_rows, _group_by_columns[i]);
+            result_chunk->append_column(std::move(res), slots[i]->id());
+        } else {
+            result_chunk->append_column(_group_by_columns[i], slots[i]->id());
+        }
+    }
+
+    if (!_agg_fn_ctxs.empty()) {
+        DCHECK(!_group_by_columns.empty());
+
+        RETURN_IF_ERROR(evaluate_agg_fn_exprs(input_chunk));
+
+        const auto num_rows = _group_by_columns[0]->size();
+        Columns agg_result_column = _create_agg_result_columns(num_rows, true);
+        for (size_t i = 0; i < _agg_fn_ctxs.size(); i++) {
+            size_t id = _group_by_columns.size() + i;
+            auto slot_id = slots[id]->id();
+
+            if (_is_merge_funcs[i] || use_intermediate_as_input) {
```

Comment:
add some comment to explain when to convert raw input format into intermediate format and when to just forward intermediate format.

Reply:
@stdpain add some comment